### PR TITLE
RFR: Reading Olympics Pages

### LINKED ISSUE #60: FE: Build out pages for Reading Olympics, and the per grade view
**Is your feature request related to a problem? Please describe.**
We have the page for an individual book. However we don't have the page that shows the list of books, or lets viewers choose which grade's list they would like to see.

**Describe the solution you'd like**
See sketches: https://1drv.ms/u/s!ApBGxqM8OyCegzcOaeP0IPpWttlB

Specifically these two pages:
- Homepage > Reading Olympics
- Homepage > Reading Olympics > Middle/Intermediate

### LINKED ISSUE #70: FE: Modify Catalog so that each card is clickable
**Is your feature request related to a problem? Please describe.**
`Catalog` renders `Card`s but we can't currently add an `onClick` function to it

**Describe the solution you'd like**
Dig into reactstrap to see how to have a clickable Card

### PR DESCRIPTION
**Pull request by XuanLi-Leong**
## Status: 

:rocket: Ready
Will need to wait on BE endpoints, but please review code first

## Description
* `ROHome` at `/ReadingOlympics`: Main page for Reading Olympics. Has 2 main outgoing links: to current year's middle school book list, and to current year's intermediate book list
* `ROYearGradeView` at `/ReadingOlympics/:year/:grade`: View catalog of books for a specified year & grade
* `ROYearView` at `/ReadingOlympics/year/:year`: View catalog of books for a specified year

Related issues: Closes #60. Closes #70 

## Todos

- [x] Check the return format of books, and convert to something that the catalog can understand
- [x] Fix the 2-column layout of `ROHome` so that the columns don't overflow page width
- [x] Style the "buttons" on `ROHome`
- [x] Map URL params to human readable strings
- [x] Use FA for default image
- [x] Get list of years and map them out in Archive card
- [ ] Need to import and render `Search` component (jk this is not built yet)
- [x]  Waiting on BE to build out `/books` API -- test the integration
- [x]   Waiting on BE to build out `/years` archive API -- test the integration --- post merge

### RO Main Page
image

### RO YearGrade View
image

### RO Year View
image

### CODE REVIEW

File: frontend/src/App.js [hunk 1]
```diff
@@ -21,6 +24,13 @@ class App extends Component {
           <Route path="/parents" component={ParentsPage} />
           <Route path="/questions" component={TestPage} />
           <Route path="/logout" component={Logout} />
+          <Route exact path="/ReadingOlympics" component={ROHome} />
+          <Route
+            exact
+            path="/ReadingOlympics/:year/:grade(middle|intermediate)"
```
Comment (Davidw1339) [inline frontend/src/App.js]: cool react router usage :arrow_double_up:

File: frontend/src/components/Catalog.js [hunk 2]
```diff
@@ -39,13 +39,14 @@ class Catalog extends Component {
 
   changePagePrev = () =>
     this.setState(state => {
-      return { page: state.page == 1 ? state.page : state.page - 1 };
+      return { page: state.page === 1 ? state.page : state.page - 1 };
```
Comment (Davidw1339) [inline frontend/src/components/Catalog.js]: oh thanks for catching this :goal_net:

File: frontend/src/pages/ReadingOlympics/ROHome.js [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+import Homefeed from '../../components/Homefeed';
+import Exercises from '../../components/Exercises';
+import Readings from '../../components/Readings';
+import Games from '../../components/Games';
+import Results from '../../components/Results';
+import '../../styles/ReadingOlympics.scss';
+import { getROArchiveYears } from '../../utils/api';
+
+const pathToMiddle = 'middle';
+const pathToIntermediate = 'intermediate';
```
Comment (Davidw1339) [inline frontend/src/pages/ReadingOlympics/ROHome.js]: what is the difference between middle and intermediate?

File: frontend/src/components/Catalog.js (hunk 2 above)
Comment (XuanLi-Leong) [reply frontend/src/components/Catalog.js]: thank npm ✨

File: frontend/src/pages/ReadingOlympics/ROHome.js (hunk 3 above)
Comment (XuanLi-Leong) [reply frontend/src/pages/ReadingOlympics/ROHome.js]: resolved offline --- Reading Olympics has 2 categories -- intermediate, and middle school

File: frontend/src/App.js (hunk 1 above)
Comment (josh-byster) [reply frontend/src/App.js]: agreed 👍

File: frontend/src/utils/api.js [hunk 4]
```diff
@@ -51,4 +51,61 @@ function search(searchString) {
   // }).then(res => res.json());
 }
 
-export { register, login, search };
+function getBooksByYearGrade({ year, grade }) {
+  if (year === null && grade === null) {
+    console.log(
+      'getBooksByYearGrade did not receive a year or grade. You should use getBooks instead'
```
Comment (josh-byster) [inline frontend/src/utils/api.js]: Could there be any edge case that causes year and grade to be null (i.e. is there FE validation that prevents this method from being called?)

File: frontend/src/utils/formatHelpers.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+function URLParamToString(param) {
+  if (param === 'middle') {
+    return 'Middle School';
```
Comment (josh-byster) [inline frontend/src/utils/formatHelpers.js]: Nice extracting this into a new file :)

File: frontend/src/utils/api.js (hunk 4 above)
Comment (XuanLi-Leong) [reply frontend/src/utils/api.js]: This function's parameters are only grabbed from URL params, so probably not haha

**Pull request merged**